frontend/src/scenes/organization/Members/index.tsx

```diff
@@ -1,56 +1,147 @@
+                                                        <CrownFilled style={{ marginRight: '0.5rem' }} />
+                                                        Pass on organization ownership
+                                                    </>
+                                                ) : listLevel > level ? (
+                                                    <>
+                                                        <UpOutlined style={{ marginRight: '0.5rem' }} />
+                                                        Promote to {organizationMembershipLevelToName.get(listLevel)}
+                                                    </>
+                                                ) : (
+                                                    <>
+                                                        <DownOutlined style={{ marginRight: '0.5rem' }} />
+                                                        Demote to {organizationMembershipLevelToName.get(listLevel)}
+                                                    </>
+                                                )}
+                                            </a>
+                                        </Menu.Item>
+                                    )
+                            )}
+                        </Menu>
+                    }
+                >
+                    {levelButton}
+                </Dropdown>
+            )
+        },
+        [user, currentMembershipLevel]
+    )
+
     const ActionsComponent = useCallback(
-        (_text, member) => {
+        (_, member: OrganizationMemberType) => {
             function handleClick(): void {
                 confirm({
                     title: `${
                         member.user_id == user.id ? 'Leave' : `Remove ${member.user_first_name} from`
-                    } organization ${user.organization.name}?`,
+                    } organization ${user.organization?.name}?`,
                     icon: <ExclamationCircleOutlined />,
-                    okText: 'Delete',
+                    okText: member.user_id == user.id ? 'Leave' : 'Remove',
                     okType: 'danger',
                     cancelText: 'Cancel',
                     onOk() {
                         removeMember(member)
+                        if (member.user_id == user.id) location.reload()
                     },
                 })
             }
 
             return (
                 <div>
-                    <a className="text-danger" onClick={handleClick}>
-                        {member.user_id !== user.id ? (
-                            <DeleteOutlined title="Remove Member" />
-                        ) : (
-                            <LogoutOutlined title="Leave Organization" />
+                    {member.level !== OrganizationMembershipLevel.Owner &&
+                        (member.level < currentMembershipLevel || member.user_id === user.id) && (
```

Comment:
Extract function here, give it a name, extract it to a file.

e.g. `canDeleteOrganizationMember(member, user)` - this is way too complicated to do inline this way.

This applies to other direct access of `currentMembershipLevel` in this file as well as when a programmer is reading this component they don't want to be thinking about access levels, but whether user has permission to do this/that.

This way there's less chance as the semantics of team membership change we'll forget to update a random component here-or-there.